app/code/Magento/NewsletterApi/Api/Data/NewsletterSubscriptionInterface.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NewsletterApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+use Magento\NewsletterApi\Api\Data\NewsletterSubscriptionExtensionInterface;
+
+/**
+ * Newsletter Subscription Interface
+ *
+ * @api
+ */
+interface NewsletterSubscriptionInterface extends ExtensibleDataInterface
+{
+    /**#@+
+     * Constants for keys of data array. Identical to the name of the getter in snake case
+     */
+    const SUBSCRIPTION_ID = 'subscription_id';
+    const EMAIL = 'email';
+    /**#@-*/
+
+    /**
+     * Set ID
+     *
+     * @param int $id
+     * @return \Magento\NewsletterApi\Api\Data\NewsletterSubscriptionInterface
```

Comment:
setter functions should return void

That's a good practice for CQS and CQRS design principles